ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/AuthService.scala

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.v1
+
+import java.util.concurrent.CompletionStage
+
+/** An interface for authorizing the ledger API access to a participant.
+  *
+  * The AuthService is responsible for converting request metadata (such as
+  * the HTTP headers) into a set of [[Claims]].
+  * These claims are then used by the ledger API server to check whether the
+  * request is authorized.
+  *
+  * For example, a participant could:
+  * - Ask all ledger API users to attach an `Authorization` header
+  *   with a JWT token as the header value.
+  * - Implement `decodeMetadata()` such that it reads the JWT token
+  *   from the corresponding HTTP header, validates the token,
+  *   and converts the token payload to [[Claims]].
+  */
+trait AuthService {
+
+  /** Converts gRPC request metadata into a set of [[Claims]].
+    *
+    *  @param headers All HTTP headers attached to the request.
+    *
+    */
+  def decodeMetadata(headers: io.grpc.Metadata): CompletionStage[Claims]
```

Comment:
Note: this introduces a GRPC dependency to the participant state. The `io.grpc.Metadata` class is useful for dealing with HTTP headers, please let me know if you think converting it into something plain, like a `List[(String, BinaryBlob)]` is a better idea.